### PR TITLE
Return url strings in response to `plek.Find`

### PR DESCRIPTION
**Pull request by vinayvinay**
outcome of a [discussion](https://github.com/alphagov/publishing-api/pull/29#discussion-diff-25501083) on another PR.

publishing-api, one of the first consuming apps of plek.go, finds it easier to work with string URLs returned from plek, and doesn't have a need for `URL` objects to be returned.

build failure's unrelated, ci-slave-2.internal has no space left.

### CODE REVIEW

File: go/plek.go [hunk 1]
```diff
@@ -77,53 +76,47 @@ func New(parentDomain string) Plek {
...
 		if _, ok := err.(*EnvVarMissing); ok {
-			return nil, nil
+			return "", nil
 		}
-		return nil, err
+		return "", err
 	}
-	return url, nil
+	return urlString, nil
 }
```
Comment (alext) [inline go/plek.go]: Given that the only error that can possibly be returned from `readEnvVarURL` is now an `EnvVarMissing` which we're handling explicitly, there's no reason for this function to have an error return value any more.  Would be cleaner to remove this.

Comment (vinayvinay) [inline go/plek.go]: @alext: fair point, fixed!

File: go/plek.go [hunk 2]
```diff
@@ -43,14 +43,10 @@ var httpDomains = map[string]bool{
...
-// URL of the service.  If it can't be parsed by url.Parse, an EnvVarURLInvalid
-// error will be returned.
-func Find(hostname string) (*url.URL, error) {
-	overrideURL, err := serviceURLFromEnvOverride(hostname)
-	if err != nil {
-		return nil, err
-	}
-	if overrideURL != nil {
+// URL of the service.
+func Find(hostname string) (string, error) {
```
Comment (alext) [inline go/plek.go]: This now no longer needs to return an error either.

File: go/plek.go [hunk 3]
```diff
@@ -77,53 +73,47 @@ func New(parentDomain string) Plek {
...
 		"PLEK_SERVICE_%s_URI",
 		strings.ToUpper(strings.Replace(serviceName, "-", "_", -1)),
 	)
-	url, err := parseEnvVarURL(varName)
+	urlString, err := readEnvVarURL(varName)
 	if err != nil {
 		if _, ok := err.(*EnvVarMissing); ok {
-			return nil, nil
+			return ""
 		}
```
Comment (alext) [inline go/plek.go]: This conditional is now redundant...  Might be worth replacing with a comment clarifying that the only error that can be returned here is an `EnvVarMissing`.

**Comment by vinayvinay** [comment]
@alext: fixed, these seem like usual teething troubles of a rubyist migrating to Go :)

**Pull request merged**